### ISSUE TITLE
Blueprint for G2/3OTA

### ISSUE THREAD

**Issue by ghost** [Potato42/kacpaw-game#1]
1. The user comes in and creates the account. Their username is returned but the password is encrypted with RSA, so anyone can encode a password and only the bot can decode it.
2. The bot decrypts the password with the private key, and hashes it before adding it to the program. This password is used as an encryption key.
3. The user logs in. The password they enter is stored in a variable `passwordInput` or the like. The user places their guess, which is encrypted with the password. Random junk characters are added to the start and end of the message to make it hard to decode the message (e.g. `mfa973|1.618|m9ao1p--}`). This is posted in the tips and thanks.
4. The bot reads this and decrypts it with its knowledge of the user's password. The guess is added to the database in both the program (encoded of course) and the bot's database.
5. On the selected day, the bot reads out all guesses, averages them, multiples by 2/3, and ranks the guesses. Each user account gets encrypted data added indicating what place they got.
6. The user logs in. 

``` javascript
if (hash(passwordInput) === user.hashedPassword) {
    var message = decode(userPlaceData, passwordInput);
    display(message);
}
```

**Comment by benburrill** [comment]
Your password generation idea makes a lot more sense when explained in this detail, and it does sound like a pretty good alternative to the quickly-edited-comment-reply.  
I've decided to create a [separate repo](https://github.com/Potato42/G2-3OTA) for G2/3OTA because it is certainly headed in a different direction than what I was thinking about and where the code for this repo is slowly heading, which would be something more along the lines of what [this guy](https://www.khanacademy.org/profile/MaxP2000/projects) sometimes moderates, except less complicated and probably in space.  But it would involve conquest and diplomacy on a map.  
However, because G2/3OTA would be simpler project (which is good because I don't really know what I'm doing) and is much better defined than my rough idea, I think we should work on that first.

From my understanding, G2/3OTA does not require the program itself to change, it would simply provide an interface for creating a password and encrypting it, as well as guesses.  So I don't really understand why we would have the code you put at the end.  However, if this is the case, it means that the front end is less tied to the back end, making it simpler to start working on (we don't need jinja2 and some specific json structure).  If you want, you could try implementing some part of the interface you want in the `program.html` file I made and send a PR.  I'll probably start working a bit on the back end tomorrow.

Anyway, thanks for your continued interest and support with my project.  :)

**Comment by ghost** [comment]
I want people to be able to see their guesses and edit them if needed.

**Comment by ghost** [comment]
The real reason I wanted the password thing is training for doing some role-play game, for example we could do a simple game involving a spy that gets to eliminate players, and you get one chance to guess at the spy...

**Comment by benburrill** [comment]
That's an interesting idea.  I hadn't really thought about making a game like that, but it would be fun.

**Comment by ghost** [comment]
Well, once we get 2/3 Average going, we can start building greater things.

I just finished Codecademy's python course. But I still don't like one thing — you can only seem to write to the end of the file? I don't like that.

**Comment by benburrill** [comment]
You can always use `seek` to move your "cursor" around a file (see [this](https://docs.python.org/3.4/tutorial/inputoutput.html#methods-of-file-objects)).  It's not really that common to want to do that though...

**Comment by ghost** [comment]
So I guess before I fork this I want to know what I'm doing...
First of all, let's say my bot account is user `@dozenalBot` and password `pswrd1728` (obviously it's not that!), where do I tell the program that's the usr and pswrd?

**Comment by benburrill** [comment]
First, you don't need to bother with forking it, just `git clone` it.  Second, don't use kacpaw-game.  Not only do you need to worry about logging in, I also completely broke it because I'm in the (not very active) process of rewriting it.
Instead, use the G2/3OTA game (`git clone https://github.com/Potato42/G2-3OTA.git`), and follow the instructions to run it.  You currently don't need to log in since I haven't really done anything with the game, it's just a simple kacpaw program that looks up info about users and programs and prints it to stdout.

**Comment by ghost** [comment]
I thought the idea of the G2/3OTA game was that people comment and the three guesses closest to 2/3 of the average are given a reward, such as compliments or helpful feedback on 5, 3, 1 of their programs.

**Comment by benburrill** [comment]
Yes, I just haven't gotten around to writing any of that yet, so it's just a simple kacpaw test program right now.  However, the mechanisms for running it will stay about the same (apart from logging in), so I thought that creating a simple, unrelated program would make it easy to test and learn about similar projects.

**Comment by ghost** [comment]
Wow. So..... what's been new? Any advancements?

**Comment by benburrill** [comment]
I've been working on some other stuff and haven't really felt like working on this.  Hopefully I'll get back to working on this somewhat soon.  I'll let you know if I make any significant progress.
